src/freenet/node/OpennetManager.java

```diff
@@ -278,6 +286,41 @@ public OpennetManager(Node node, NodeCryptoConfig opennetConfig, long startupTim
 		    peersLRUByDistance.put(l, new LRUQueue<OpennetPeerNode>());
 		oldPeers = new LRUQueue<OpennetPeerNode>();
 		announcer = (enableAnnouncement ? new Announcer(this) : null);
+
+		int sortOrder = 0;
+		final SubConfig sConfig = node.config.get("node.opennet");
+
+		sConfig.register("excludedCountries",  "", sortOrder++, false, false, "Node.opennetExcludedCountriesShort","Node.opennetExcludedCountriesLong",
+		new StringCallback() {
+
+			@Override
+			public String get() {
+				StringBuilder joined = new StringBuilder();
+				for (String excludedCountry : excludedCountries) {
+					if (joined.length() > 0) {
+						joined.append(',');
+					}
+					joined.append(excludedCountry).append(',');
+				}
+				return joined.toString();
+			}
+
+			@Override
+			public void set(String newValue) /* throws InvalidConfigValueException, NodeNeedRestartException*/ {
+				excludedCountries.clear();
+				if ((newValue != null) && !newValue.equals("")) {
+					for (String excludedCountry : newValue.split(",")) {
+						excludedCountries.add(excludedCountry.toUpperCase());
```

Comment:
Further, please validate whether the user's input country codes are valid before storing them in the configuration.